vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -479,23 +479,30 @@ testNetworkSettings() {
 # Ensures that the content /etc/profile.d/umask.sh is correct, per code in
 # <repo-root>/parts/linux/cloud-init/artifacts/cis.sh
 testUmaskSettings() {
-  local test="testUmaskSettings"
-  local settings_file=/etc/profile.d/umask.sh
-  local expected_settings_file_content='umask 027'
-  echo "$test:Start"
-
-  # If the settings file exists, it must just be a single line that sets umask properly.
-  if [[ -f "${settings_file}" && -s "${settings_file}" ]]; then
-    echo "${test}: Checking that ${settings_file} contains '${expected_settings_file_content}'"
-
-    if ! diff "${settings_file}" <(echo "${expected_settings_file_content}") ; then
-      err $test "The content of the file '${settings_file}' must exactly match '${expected_settings_file_content}'. See above for differences"
+    local test="testUmaskSettings"
+    local settings_file=/etc/profile.d/umask.sh
+    local expected_settings_file_content='umask 027'
+    echo "$test:Start"
+
+    # If the settings file exists, it must just be a single line that sets umask properly.
+    if [[ -f "${settings_file}" ]]; then
+        echo "${test}: Checking that the contents of ${settings_file} is exactly '${expected_settings_file_content}'"
+
+        # Command substitution (like file_contents=$(cat "${settings_file}")) strips trailing newlines, so we use mapfile instead.
+        # This creates an array of the lines in the file, and then we join them back together by expanding the array into a single string.
+        local file_contents_array=()
+        mapfile <"${settings_file}" file_contents_array
```

Comment:
do you even need mapfile vs cat here for simple string?

I also suspect `git diff --no-index` may work?

Reply:
We need to get trailing whitespace, which command-substitution with `cat` will trim. `git diff` is a good idea, though it feels weird to use `git` to do a non-git thing if that makes sense?